src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -1,33 +1,153 @@
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        ArgumentMultimap argMultimap = ArgumentTokenizer
+                .tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_ROOM_NUMBER, PREFIX_TAG);
+
+        if (!argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        Optional<String> name = parseName(argMultimap.getValue(PREFIX_NAME).orElse(null));
+        Optional<String> contactNumber = parseContactNumber(argMultimap.getValue(PREFIX_PHONE).orElse(null));
+        Optional<String> roomNumber = parseRoomNumber(argMultimap.getValue(PREFIX_ROOM_NUMBER).orElse(null));
+        Optional<List<String>> tagList = parseTags(argMultimap.getAllValues(PREFIX_TAG));
+
+        Predicate<Person> combinedPredicate = p -> true;
+
+        if (name.isPresent()) {
+            combinedPredicate = combinedPredicate
+                    .and(new NameContainsKeywordsPredicate(Arrays.asList(name.get().split("\\s+"))));
+        }
+        if (contactNumber.isPresent()) {
+            combinedPredicate = combinedPredicate.and(new PhonePredicate(contactNumber.get()));
+        }
+        if (roomNumber.isPresent()) {
+            combinedPredicate = combinedPredicate.and(new RoomNumberPredicate(roomNumber.get()));
+        }
+        if (tagList.isPresent()) {
+            combinedPredicate = combinedPredicate.and(new TagContainsKeywordsPredicate(tagList.get()));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        return new FindCommand(combinedPredicate);
+    }
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+    /**
+     * Parses and validates the name extracted from the input arguments.
+     * Returns an Optional containing the name if it is valid, otherwise throws ParseException.
+     * @param name The name to validate.
+     * @return Optional Containing the validated name, or empty if no name is provided.
+     * @throws ParseException If the name does not meet the validation criteria.
+     */
+    private Optional<String> parseName(String name) throws ParseException {
+        if (name != null && !name.trim().isEmpty()) {
+            if (!Name.isValidName(name.trim())) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, Name.CHAR_MESSAGE_CONSTRAINTS));
+            }
+            return Optional.of(name.trim());
+        }
+        return Optional.empty();
     }
 
+    /**
+     * Parses and validates the contact number from the input arguments.
+     * Returns an Optional containing the contact number if it is valid, otherwise throws ParseException.
+     * @param contactNumber The contact number to validate.
```

Comment:
Empty line between description and parameter section.

Reply:
what do you mean?